assessment.md

```diff
@@ -147,39 +147,77 @@ TODO: add description for each assessment level for this skill.
 
 * Requirements engineering
 
-TODO: add description for each assessment level for this skill.
+* :flushed: = Fails to adds any requirements for the program and shows no knowledge of what the program is doing
+* :worried: = Adding requirements, but not making them user-focused.
+* :neutral_face: = Adding user-focused requirements, but not having appropriate estimates for those requirements.
+* :blush: = Adding user-focused requirements with appropriate estimates, but the individual requirements involve a multitude of different tasks instead of just one.
+* :laughing: = Adding user-focused requirements with appropriate estimates, and the individual requirments involve only one task.
 
 * Software architecture
 
-TODO: add description for each assessment level for this skill.
+* :flushed: = The program demonstrates no understanding or implementation of software architecture.
+* :worried: = The program displays experience with software architecture, but there are flow and design issues.
+* :neutral_face: = The program displays experience and understanding of software architecture with minimal flow and design issues.
+* :blush: = the program displays an extensive knowledge of software architecture theory and underlying experience with no flow or design issues.
+* :laughing: = the program displays an extensive knowledge of software theory and underlying experience as a means of conducting and managing the design/flow. The student develops concepts and is able to help other team members with the architecture of their software.
 
 * Software design
 
-TODO: add description for each assessment level for this skill.
+* :flushed: = Creates no design for the software.
+* :worried: = Creates a design, but doesn't support the software's goals.
+* :neutral_face: = Creates a design that supports the software's goals, but perhaps not an efficient design.
+* :blush: = Creates an efficient, robust design that supports the software's goals.
+* :laughing: = In addition to creating an efficient, robust design, constantly thinks about how the design could be improved.
 
 * Software documentation
 
-TODO: add description for each assessment level for this skill.
+* :flushed: = No explanation of what the software does, no solution to the problem presented, and no example of where the code be used is included
+* :worried: = A short explanation, no solution to the problem, and no example is included
+* :neutral_face: = A good explanation, a short solution to the problem, and a short example is included
+* :blush: = A detailed explanation, a solution to the problem, a short example is included
+* :laughing: = A detailed explanation, a detailed solution to the problem, a interactive example is included
```

Comment:
This section should also address the variety of software documentation, docstrings, type annotations, README, feature docs, etc.